marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -695,24 +674,25 @@ instance SemiArbitrary Integer where
 
 
 instance Arbitrary (Value Observation) where
-  arbitrary =
-    frequency
-      [
-        ( 8, AvailableMoney <$> arbitrary <*> arbitrary)
-      , (14, Constant <$> arbitrary)
-      , ( 8, NegValue <$> arbitrary)
-      , ( 8, AddValue <$> arbitrary <*> arbitrary)
-      , ( 8, SubValue <$> arbitrary <*> arbitrary)
-      , ( 8, MulValue <$> arbitrary <*> arbitrary)
-      , ( 8, DivValue <$> arbitrary <*> arbitrary)
-      , (10, ChoiceValue <$> arbitrary)
-      , ( 6, pure TimeIntervalStart)
-      , ( 6, pure TimeIntervalEnd)
-      , ( 8, UseValue <$> arbitrary)
-      , ( 8, Cond <$> arbitrary <*> arbitrary <*> arbitrary)
-      ]
+  arbitrary = sized \size ->
+    if size <= 0
+      then frequency -- size 0 produces terminal values.
+        [ (4, AvailableMoney <$> arbitrary <*> arbitrary)
+        , (7, Constant <$> arbitrary)
+        , (5, ChoiceValue <$> arbitrary)
+        , (3, pure TimeIntervalStart)
+        , (3, pure TimeIntervalEnd)
+        , (4, UseValue <$> arbitrary)
+        ]
+      else oneof -- size > 0 produces compound values
+        [ resize (pred size) $ NegValue <$> arbitrary
+        , resize (size `quot` 2) $ AddValue <$> arbitrary <*> arbitrary
+        , resize (size `quot` 2) $ SubValue <$> arbitrary <*> arbitrary
+        , resize (size `quot` 2) $ MulValue <$> arbitrary <*> arbitrary
+        , resize (size `quot` 2) $ DivValue <$> arbitrary <*> arbitrary
+        , resize (size `quot` 3) $ Cond <$> arbitrary <*> arbitrary <*> arbitrary
+        ]
   shrink (AvailableMoney a t) = [AvailableMoney a' t | a' <- shrink a] ++ [AvailableMoney a t' | t' <- shrink t]
-  shrink (Constant x)         = Constant <$> shrink x
```

Comment:
Shrinking an integer is unnecessary.

Reply:
It can point out where the error is, if the error is because of the number, then it may be the one that doesn't shrink

Reply:
I will restore the shrinking of integers